store/mongo/recording_writer.go

```diff
@@ -47,7 +47,12 @@ func sendRecordingMessage(data []byte, sessionID string, w io.Writer) (int, erro
 	if err != nil {
 		return 0, err
 	} else {
-		return w.Write(messagePacked)
+		n, e := w.Write(messagePacked)
+		if e != nil {
+			return n, e
+		} else {
+			return len(data), nil
+		}
```

Comment:
To me it seems like `w` is not implementing the `io.Writer` interface correctly

> Write writes len(p) bytes from p to the underlying data stream. It returns the number of bytes written from p (0 <= n <= len(p)) and any error encountered that caused the write to stop early. Write must return a non-nil error if it returns n < len(p). 
> [...]
> ```go
> type Writer interface {
>     Write(p []byte) (n int, err error)
> }
> ```

Reply:
discussed over slack. it is essentially correct -- we are fixing the Writer interface, only onle level up. w is calling WriteSessionRecords and there we are positng to nats -- we have no control over what is encapsulated inside the packedmessage.